radixdlt-java/src/main/java/com/radixdlt/client/application/RadixApplicationAPI.java

```diff
@@ -355,6 +356,62 @@ public TokenClassReference getNativeTokenRef() {
 		return getTokenClass(getNativeTokenRef());
 	}
 
+	/**
+	 * Returns a never ending hot observable of the actions performed at a given address.
+	 * If the given address is not currently being pulled this will pull for atoms in that
+	 * address automatically until the observable is disposed.
+	 *
+	 * @param actionClass the Action class
+	 * @param address the address to retrieve the state of
+	 * @param <T> the Action class
+	 * @return a hot observable of the actions at the given address
+	 */
+	public <T> Observable<T> getActions(Class<T> actionClass, RadixAddress address) {
+		final Observable<AtomObservation> atomsPulled = ledger.getAtomPuller() != null
+			? ledger.getAtomPuller().pull(address)
+			: Observable.never();
+		Observable<AtomObservation> auto = atomsPulled.publish()
+			.refCount(2);
+		Disposable d = auto.subscribe();
+
+		return this.getActionStore(actionClass).getActions(address, identity)
+			.map(actionClass::cast)
+			.publish()
+			.autoConnect()
+			.doOnSubscribe(disposable -> auto.subscribe().dispose())
+			.doOnError(e -> d.dispose())
+			.doOnDispose(d::dispose)
+			.doOnComplete(d::dispose);
+	}
+
+	/**
+	 * Returns a never ending hot observable of the state of a given address.
+	 * If the given address is not currently being pulled this will pull for atoms in that
+	 * address automatically until the observable is disposed.
+	 *
+	 * @param stateClass the ApplicationState class
+	 * @param address the address to retrieve the state of
+	 * @param <T> the ApplicationState class
+	 * @return a hot observable of a state of the given address
+	 */
+	public <T extends ApplicationState> Observable<T> getState(Class<T> stateClass, RadixAddress address) {
+		final Observable<AtomObservation> atomsPulled = ledger.getAtomPuller() != null
+				? ledger.getAtomPuller().pull(address)
+				: Observable.never();
+		Observable<AtomObservation> auto = atomsPulled.publish()
+			.refCount(2);
+		Disposable d = auto.subscribe();
+
+		return this.getStore(stateClass).getState(address)
+			.map(stateClass::cast)
+			.publish()
+			.autoConnect()
+			.doOnSubscribe(disposable -> auto.subscribe().dispose())
+			.doOnError(e -> d.dispose())
```

Comment:
We seem to lose error information here.

Reply:
Don't need the error though for this lambda, it just gets propagated downstream to the correct handler.